modules/ximgproc/src/brightedges.cpp

```diff
@@ -0,0 +1,233 @@
+			input[iedge.cols * (row + 1) + (col + 2)];
+		if (line == 0) lines += 1;
+
+		line =
+			input[iedge.cols * (row - 2) + (col)] +
+			input[iedge.cols * (row - 1) + (col)] +
+			input[iedge.cols * (row + 1) + (col + 1)] +
+			input[iedge.cols * (row + 2) + (col + 1)];
+		if (line == 0) lines += 1;
+
+		line =
+			input[iedge.cols * (row - 2) + (col - 1)] +
+			input[iedge.cols * (row - 1) + (col - 1)] +
+			input[iedge.cols * (row + 1) + (col)] +
+			input[iedge.cols * (row + 2) + (col)];
+		if (line == 0) lines += 1;
+
+		line =
+			input[iedge.cols * (row - 2) + (col - 2)] +
+			input[iedge.cols * (row - 1) + (col - 2)] +
+			input[iedge.cols * (row - 2) + (col - 1)] +
+			input[iedge.cols * (row - 1) + (col - 1)] +
+			input[iedge.cols * (row + 1) + (col + 1)] +
+			input[iedge.cols * (row + 1) + (col + 2)] +
+			input[iedge.cols * (row + 2) + (col + 1)] +
+			input[iedge.cols * (row + 2) + (col + 2)];
+		if (line < weight) lines += 1;
+
+		// Near vertical 
+		line =
+			input[iedge.cols * (row - 2) + (col)] +
+			input[iedge.cols * (row - 1) + (col)] +
+			input[iedge.cols * (row - 2) + (col - 1)] +
+			input[iedge.cols * (row - 2) + (col + 1)] +
+			input[iedge.cols * (row + 1) + (col)] +
+			input[iedge.cols * (row + 2) + (col)] +
+			input[iedge.cols * (row + 2) + (col + 1)] +
+			input[iedge.cols * (row + 2) + (col - 1)];
+		if (line < weight) lines += 1;
+		// Near diagonal top right to bottom left 
+
+		line =
+			input[iedge.cols * (row - 2) + (col + 2)] +
+			input[iedge.cols * (row - 1) + (col + 1)] +
+			input[iedge.cols * (row - 2) + (col - 1)] +
+			input[iedge.cols * (row - 1) + (col + 2)] +
+			input[iedge.cols * (row + 1) + (col - 1)] +
+			input[iedge.cols * (row + 2) + (col - 2)] +
+			input[iedge.cols * (row + 2) + (col - 1)] +
+			input[iedge.cols * (row + 1) + (col - 2)];
+		if (line < weight) lines += 1;
+
+		// Near horizontal 
+		line =
+			input[iedge.cols * (row)+(col - 2)] +
+			input[iedge.cols * (row)+(col - 1)] +
+			input[iedge.cols * (row - 1) + (col - 2)] +
+			input[iedge.cols * (row + 1) + (col - 2)] +
+			input[iedge.cols * (row)+(col + 1)] +
+			input[iedge.cols * (row)+(col + 2)] +
+			input[iedge.cols * (row + 1) + (col + 2)] +
+			input[iedge.cols * (row - 1) + (col + 2)];
+		if (line < weight) lines += 1;
+
+		if (line == 1) return 0;
+
+		int surround = input[iedge.cols * (row - 1) + (col - 1)] +
+			input[iedge.cols * (row - 1) + (col)] +
+			input[iedge.cols * (row - 1) + (col + 1)] +
+			input[iedge.cols * (row)+(col - 1)] +
+			input[iedge.cols * (row)+(col + 1)] +
+			input[iedge.cols * (row + 1) + (col - 1)] +
+			input[iedge.cols * (row + 1) + (col)] +
+			input[iedge.cols * (row - 1) + (col + 1)];
+
+		if (surround == 8 * 255) return 255;
+		if (surround == 0) return 255;
+
+		return input[iedge.cols * (row)+(col)];
+
+
+	}
+	
+	int contrastEdges(Mat &minput, Mat &mouput, int contrast) {
+
+		unsigned char *input = (unsigned char*)(mouput.data);
+
+		Mat mwork = minput.clone();
```

Comment:
Cloning is excessive here. It would be better to just create the matrix of given size and type.
i.e. `Mat mwork(minput.size(), minput.type(), Scalar(255));`

Reply:
Replaced clone thank you.